pkg/generate/generate.go

```diff
@@ -47,10 +47,15 @@ func (c *Controller) applyGenerate(resource unstructured.Unstructured, gr kyvern
 	policy, err := c.pLister.Get(gr.Spec.Policy)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			if err := c.client.DeleteResource(gr.Spec.Resource.Kind, gr.Spec.Resource.Namespace, gr.Spec.Resource.Name, false); err != nil {
-				logger.V(4).Info("Generated resource is deleted")
-				return nil, err
+			for _, e := range policy.Spec.Rules {
```

Comment:
Here returns policy not found error, we cannot check synchronize flag using policy. It should be available in resource label.

Reply:
one policy can have multiple role and synchronize flag is for rule